Classes/Hooks/Interfaces/RegistrationEmail.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OliverKlee\Seminars\Hooks\Interfaces;
+
+/**
+ * Use this interface for hooks concerning the registration emails.
+ *
+ * It supersedes the deprecated `RegistrationEmailHookInterface` interface.
+ *
+ * Customize emails before they are sent.
+ *
+ * @author Pascal Rinker <projects@jweiland.net>
+ * @author Michael Kramer <m.kramer@mxp.de>
+ */
+interface RegistrationEmail extends Hook
+{
+    /**
+     * Modifies the attendee "Thank you" email just before it is sent.
+     *
+     * You may modify receiver or sender as well as subject and body of the email.
+     *
+     * @param \Tx_Oelib_Mail $mail
+     * @param \Tx_Seminars_Model_Registration $registration
+     * @param string $emailReason Possible values:
+     *          - confirmation
+     *          - confirmationOnUnregistration
+     *          - confirmationOnRegistrationForQueue
+     *          - confirmationOnQueueUpdate
+     *
+     * @return void
+     */
+    public function modifyAttendeeEmail(
+        \Tx_Oelib_Mail $mail,
+        \Tx_Seminars_Model_Registration $registration,
+        string $emailReason
+    );
+
+    /**
+     * Modifies the attendee "Thank you" email body just before the subpart is rendered to PlainText.
+     *
+     * This method is called for every confirmation mail, even if HTML emails are configured.
+     * The body of a HTML mail alwyas contains a PlainText version, too.
```

Comment:
I propose we reword this to make it clear that this hook is called for the plain text part of all HTML emails as well.